app/assets/javascripts/components/comment_list.js.jsx

```diff
@@ -0,0 +1,67 @@
+var CommentList = createReactClass({
+  propTypes: {
+    projectId: PropTypes.number,
+    currentUserId: PropTypes.number,
+    deleteBasePath: PropTypes.string,
+    deletePermission: PropTypes.bool,
+    initialComments: PropTypes.array
+  },
+
+  getInitialState: function() {
+    CommentStore.subscribe(this, this.updateComments);
```

Comment:
In general looks great! I may be misunderstanding, but I think if you do
```
this.updateComments = this.updateComments.bind(this);
```
on the line above this line, you can avoid all the effort of keeping the `parent` inside of the store and also needing to pass it into `updateComments` as an argument.

Reply:
@veryeasily It looks like my non-ES6 way of creating React components comes with autobinding, which means I don't even need the `bind` call.

https://reactjs.org/docs/react-without-es6.html#autobinding